firedrake/slate/slac/utils.py

```diff
@@ -137,6 +136,37 @@ def visit_Symbol(self, o, *args, **kwargs):
         return SymbolWithFuncallIndexing(o.symbol, o.rank, o.offset)
 
 
+def topological_sort(exprs):
+    """Returns a topologically sorted list of Slate expressions.
+    The list is generated by examining the dependencies of each
+    expression and moving nodes that have all its dependencies
+    resolved. A node is resolved once all the nodes in its dependencies
+    have been moved into the sorted list.
```

Comment:
This just described you implementation of the topo sort.  Better to just say what graph it is you are sorting.